src/main/java/se/kth/sda/skeleton/comments/CommentController.java

```diff
@@ -0,0 +1,76 @@
+package se.kth.sda.skeleton.comments;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class is responsible for mapping comments data to endpoints.
+ */
+
+@RestController
+@RequestMapping("/comments")
+public class CommentController {
+
+    @Autowired
+    CommentService commentService;
+
+    /**
+     * Retrieves all the comments made to all the posts available on the forum
+     * unless an additional parameter is added. In this case,
+     * only the comments related to the post whose id was specified as the parameter
+     * will be returned.
+     * @param postId
+     * @return List of all comments or comments associated to the specific post if postId was passed as a parameter.
+     */
+    @GetMapping("")
+    public List<Comment> getAll(@RequestParam(required = false) Long postId) {
+        if (postId.equals(null)) {
+            return commentService.getAll();
+        } else {
+            return commentService.getAllByPostId(postId);
+        }
+    }
+
+    /**
+     * Gets a comment by its id.
+     * @param id
+     * @return Comment with a specified id.
+     */
+    @GetMapping("/{id}")
```

Comment:
should it be a list of Comments returned if the id is unique and singular?

Reply:
I don't think we will use this method in the mandatory task. But it should not be a problem if left alone for now, we may probably need this method to get individual comments at some point.

Reply:
I will just accept it for now then we can change it later if needed